cpp/include/Ice/ProxyFunctions.h

```diff
@@ -103,22 +104,22 @@ namespace Ice
         }
     }
 
-    /// Downcasts a proxy after confirming the target object's type via a remote invocation.
+    /// Creates a new proxy from an existing proxy after confirming the target object's type via a remote invocation.
     /// @param proxy The source proxy (can be nullopt).
     /// @param context The request context.
-    /// @return A proxy with the requested type, or nullopt if the source proxy is nullopt or if the target object does
```

Comment:
I removed the "if the source proxy is nullopt" for several reasons:
- arguably, if the source proxy is nullopt, the target doesn't implement the request type!
- the return sentence is much more readable without this "if"
- in general, you should not make remote invocations on null proxy. We only keep this option for backwards compatibility with the old API.